tests/virt_autotest/virsh_virtual_network.pm

```diff
@@ -0,0 +1,409 @@
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+# Summary: Managing Virtual Networks with virsh included
+# the following priorities 4 types of virtual network:
+# P1 HOST bridge network
+# P2 NAT based network
+# P3 Routed network
+# P4 Isolated network
+# Maintainer: Leon Guo <xguo@suse.com>
+
+use base "consoletest";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use set_config_as_glue;
+use version_utils 'is_sle';
+
+our $virt_host_bridge = 'br0';
+sub prepare_network {
+    #Confirm the host bridge configuration file
+    my $config_path = "/etc/sysconfig/network/ifcfg-$virt_host_bridge";
+    if (script_run("[[ -f $config_path ]]") != 0) {
+        assert_script_run("ip link add name $virt_host_bridge type bridge");
+        assert_script_run("ip link set dev $virt_host_bridge up");
+        my $bridge_setup = <<'EOF';
+#!/bin/bash
+. /etc/os-release
+ACTIVE_NET=$(ip a|awk -F': ' '/state UP/ {print $2}'|head -n1)
+interface="BOOTPROTO='none'\nSTARTMODE='auto'"
+bridge="BOOTPROTO='dhcp'\nBRIDGE='yes'\nBRIDGE_FORWARDDELAY='0'\nBRIDGE_PORTS='$ACTIVE_NET'\nBRIDGE_STP='off'\nSTARTMODE='auto'"
+> /etc/sysconfig/network/ifcfg-br0.new
+echo -e $bridge >/etc/sysconfig/network/ifcfg-br0
+echo -e $interface >/etc/sysconfig/network/ifcfg-$ACTIVE_NET
+cat /etc/sysconfig/network/ifcfg-br0
+cat /etc/sysconfig/network/ifcfg-$ACTIVE_NET
+if [[ $VERSION_ID =~ '11' ]]; then
+    service libvirtd restart;
+else
+    systemctl restart network.service;
+fi
+EOF
+        script_output($bridge_setup, 900);
+    }
+}
+
+sub restore_standalone {
+    #File standalone was installed from qa_test_virtualization package
+    my $standalone_path = "/usr/share/qa/qa_test_virtualization/shared/standalone";
+    assert_script_run("source $standalone_path", 60) if (script_run("[[ -f $standalone_path ]]") == 0);
+}
+
+sub destroy_standalone {
+    #File cleanup was installed from qa_test_virtualization package
+    my $cleanup_path = "/usr/share/qa/qa_test_virtualization/cleanup";
+    assert_script_run("source $cleanup_path", 60) if (script_run("[[ -f $cleanup_path ]]") == 0);
+}
+
+sub reset_libvirt_service {
+    assert_script_run qq(echo 'log_level = 1
+    log_filters="3:remote 4:event 3:json 3:rpc"
+    log_outputs="1:file:/var/log/libvirt/libvirtd.log"' >> /etc/libvirt/libvirtd.conf);
+}
+
+sub restart_libvirtd {
+    is_sle('>11') ? systemctl 'restart libvirtd' : script_run("service libvirtd restart");
+}
+
+sub upload_libvirtd_log {
+    upload_logs "/var/log/libvirt/libvirtd.log";
```

Comment:
Not only libvirtd log is needed, you can refer to our other prjs for full log, eg /var/log/xen /var/log/libvirt (qemu log is there), system log of journalctl, etc

These are useful when failures hap

Reply:
Maybe later on #8241 @waynechen55 can move this to `lib/virsh_autotest_base.pm` base test module.

Reply:
Working on upload debug logs for libvirt virtual network.

Reply:
Had added `upload_debug_log` to upload debug logs  libvirt virtual network.